src/controllers/hibp.js

```diff
@@ -40,144 +22,64 @@ const log = mozlog('controllers.hibp')
  * @param {object} res
  */
 async function notify (req, res) {
-  if (!req.token || req.token !== AppConstants.HIBP_NOTIFY_TOKEN) {
-    const errorMessage = 'HIBP notify endpoint requires valid authorization token.'
-    throw new UnauthorizedError(errorMessage)
-  }
-  if (!['breachName', 'hashPrefix', 'hashSuffixes'].every(req.body?.hasOwnProperty, req.body)) {
-    throw new UserInputError('HIBP breach notification: requires breachName, hashPrefix, and hashSuffixes.')
-  }
-
-  const { breachName, hashPrefix, hashSuffixes } = req.body
-
-  let breachAlert = getBreachByName(req.app.locals.breaches, breachName)
-
-  if (!breachAlert) {
-    // If breach isn't found, try to reload breaches from HIBP
-    log.debug('notify', 'Breach is not found, reloading breaches...')
-    await loadBreachesIntoApp(req.app)
-    breachAlert = getBreachByName(req.app.locals.breaches, breachName)
-    if (!breachAlert) {
-      // If breach *still* isn't found, we have a real error
-      throw new Error('Unrecognized breach: ' + breachName)
-    }
-  }
-
-  const { IsVerified, Domain, IsFabricated, IsSpamList } = breachAlert
-
-  // If any of the following conditions are not satisfied:
-  // Do not send the breach alert email! The `logId`s are being used for
-  // logging in case we decide to not send the alert.
-  const emailDeliveryConditions = [
-    {
-      logId: 'isNotVerified',
-      condition: !IsVerified
-    },
-    {
-      logId: 'domainEmpty',
-      condition: Domain === ''
-    },
-    {
-      logId: 'isFabricated',
-      condition: IsFabricated
-    },
-    {
-      logId: 'isSpam',
-      condition: IsSpamList
-    }
-  ]
-
-  const unsatisfiedConditions = emailDeliveryConditions.filter(condition => (
-    condition.condition
-  ))
-
-  const doNotSendEmail = unsatisfiedConditions.length > 0
-
-  if (doNotSendEmail) {
-    // Get a list of the failed condition `logId`s
-    const conditionLogIds = unsatisfiedConditions
-      .map(condition => condition.logId)
-      .join(', ')
-
-    log.info('Breach alert email was not sent.', {
-      name: breachAlert.Name,
-      reason: `The following conditions were not satisfied: ${conditionLogIds}.`
-    })
-
-    return res.status(200).json({
-      info: 'Breach loaded into database. Subscribers not notified.'
-    })
-  }
-
   try {
-    const reqHashPrefix = hashPrefix.toLowerCase()
-    const hashes = hashSuffixes.map(suffix => reqHashPrefix + suffix.toLowerCase())
-    const subscribers = await getSubscribersByHashes(hashes)
-    const emailAddresses = await getEmailAddressesByHashes(hashes)
-    const recipients = subscribers.concat(emailAddresses)
-
-    log.info(EmailTemplateType.Notification, {
-      breachAlertName: breachAlert.Name,
-      length: recipients.length
-    })
+    if (AppConstants.NODE_ENV === 'dev') {
+      return notifyWorker(req, res)
+    } else {
+      const url = '' // TODO
```

Comment:
When a function is created in GCP, does it give out a public url? With AWS I know we use APIGateway -> functions to create a reachable service